torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -259,6 +259,38 @@ class TORCH_API RpcAgent {
   // Get the type resolver
   std::shared_ptr<TypeResolver> getTypeResolver();
 
+  static std::shared_ptr<JitFuture> toJitFuture(std::shared_ptr<FutureMessage> fm) {
+    auto jitFuture = std::make_shared<JitFuture>(at::AnyClassType::get());
+
+    fm->addCallback(
+        [jitFuture, fm]() mutable {
```

Comment:
I remember last time I worked on C++ futures API, Jeremy had a comment here: https://github.com/pytorch/pytorch/pull/40326#discussion_r449289403 about avoiding (possible) refcycles and suggesting to use a weak_ptr. 

The motivation is that sometimes, the callbacks are never run (for various reasons). In this case, we have a copy of the shared_ptr so an additional reference to it, so the pointed to memory is never freed due to the ref cycle.

Ultimately in that PR we went with using weakref instead of the shared_ptr copy, so just wanted to call it out in case we want to do that here as well.